error.go

```diff
@@ -38,10 +39,81 @@ func (e *errResponse) Response() *http.Response {
 	return e.response
 }
 
-// NewError ...
-func NewError(expectedRespCode []int, resp *http.Response) Error {
+// NewResponseError ...
+func NewResponseError(expectedRespCode []int, resp *http.Response) ResponseError {
 	return &errResponse{
 		expectedResponseCode: expectedRespCode,
 		response:             resp,
 	}
 }
+
+// NewError wraps all details we get during screening or checkr requests
+func NewError(id string, st ScreenType, err error, expectedRespCode int, resp *http.Response) *Error {
```

Comment:
@chandradeepak I don't think this is needed, right?

Reply:
this is the new one. we are using this now

Reply:
@chandradeepak ah, I understand now. Sorry, but actually, the `ResponseError` is actually good for this. Because the error is caused by calling `Checkr` and the `response` is considered as an `error`.

Reply:
@chandradeepak just use `errResponse` and I see you're identifying the `screening type` and I don't think there's a need to this but if you think it's useful, perhaps you can extend `errResponse` to have a `metadata` field and make that part of the output.

Reply:
yes i integrated Error type